### ISSUE TITLE
Can randomly split on load/save load under some unknown circumstance

### ISSUE THREAD

**Issue by Zedb0T** [open-goal/speedrunning#19]
Happens in all orbs to Vortex when save loading, but only after training.

Ideas: When you load the save maybe the flag/event for collecting all the orbs sends again or something so the split triggers a second time

Maybe there is a misaligned index, and when you complete some task it sets the incorrect location in the struct to 1, then reads the correct location as 0 and splits again etc.

Eventually I will look at this, but I am going to be out of town soon so if anyone wants to poke around on this or has any suggestions to look/test feel free.

**Comment by xTVaser** [comment]
A video recording would help

**Comment by Zedb0T** [comment]
I have found the issue, just looking at solutions now, when you load a save `(-> *game-info* money-per-level (level-task-data-index training))` gets set to 0 until the load is finished, then it gets set to the correct value

**Comment by Zedb0T** [comment]
https://clips.twitch.tv/LongSnappyWalletGingerPower-DTfRvH3u2vCFR2Ng

```2
2
2
2
2
[02:42] [warn] Failed to find texture at 256, using random (sprite)
[02:42] [warn] Failed to find texture at 832, using random (sprite)
[02:42] [warn] Failed to find texture at 848, using random (sprite)
[02:42] [warn] Failed to find texture at 768, using random (sprite)
****** CALL TO kopen() ******
pc settings file write: "C:\Users\NinjaPC\AppData\Roaming\OpenGOAL\jak1\settings/pc-settings.gc"
[000c1c04] game-save
        version: 1
        allocated-length: 252848
        length: 6736
        level-index: 1
        fuel-cell-count: 0.0000
        money-count: 2.0000
        buzzer-count: 0.0000
        completion-percentage: 0.0099
        save-time: 19:2 7/1/24
        tag[]: @ #xc1c50
          [  0]                             name [  5/  1]             0   0.0000    = "save"
          [  1]                        base-time [  0/  0]        303030   0.0000    
          [  2]                        real-time [  0/  0]        305685   0.0000    
          [  3]                        game-time [  0/  0]        303030   0.0000    
          [  4]                    integral-time [  0/  0]        300606   0.0000    
          [  5]                         continue [ 15/  1]             0   0.0000    = "training-start"
          [  6]                             life [  0/  0]    1084227584   5.0000    
          [  7]                     buzzer-total [  0/  0]             0   0.0000    
          [  8]                        fuel-cell [  0/  0]             0   0.0000    
          [  9]                 death-movie-tick [  0/  0]             2   0.0000    
          [ 10]                            money [  0/  0]    1073741824   2.0000    
          [ 11]                      money-total [  0/  0]    1073741824   2.0000    
          [ 12]                  money-per-level [ 32/  1]             0   0.0000    
          [ 13]                  level-open-list [ 32/  1]             0   0.0000    
          [ 14]                        perm-list [ 84/ 16]             0   0.0000    
          [ 15]                        task-list [116/ 16]             0   0.0000    
          [ 16]                        text-list [512/  1]             0   0.0000    
          [ 17]                        hint-list [ 25/ 32]             0   0.0000    
          [ 18]                  auto-save-count [  0/  0]             0   0.0000    
          [ 19]                     total-deaths [  0/  0]             0   0.0000    
          [ 20]                  continue-deaths [  0/  0]             0   0.0000    
          [ 21]                 fuel-cell-deaths [  0/  0]             0   0.0000    
          [ 22]                  game-start-time [  0/  0]        300000   0.0000    
          [ 23]                    continue-time [  0/  0]        300560   0.0000    
          [ 24]                       death-time [  0/  0]        300000   0.0000    
          [ 25]                         hit-time [  0/  0]        300000   0.0000    
          [ 26]            fuel-cell-pickup-time [  0/  0]        300000   0.0000    
          [ 27]                   fuel-cell-time [116/  8]             0   0.0000    
          [ 28]                 deaths-per-level [ 32/  1]             0   0.0000    
          [ 29]                 enter-level-time [ 32/  8]             0   0.0000    
          [ 30]                    in-level-time [ 32/  8]             0   0.0000    
          [ 31]                       sfx-volume [  0/  0]    1117126656  75.0000    
          [ 32]                     music-volume [  0/  0]    1115815936  65.0000    
          [ 33]                    dialog-volume [  0/  0]    1120403456 100.0000    
          [ 34]                         language [  0/  0]             0   0.0000    
          [ 35]                           screenx [  0/  0]             0   0.0000    
          [ 36]                          screeny [  0/  0]    1090519040   8.0000    
          [ 37]                        vibration [  0/  0]             1   0.0000    
          [ 38]                       play-hints [  0/  0]             0   0.0000    
          [ 39]                       video-mode [  0/  0]             1   0.0000    
          [ 40]                     aspect-ratio [  0/  0]             1   0.0000    
[02:47] [debug] [OVERLORD] LoadISOFileChunkToEE FUCV7   STR : 145408 offset 1        
[02:47] [debug] [OVERLORD] Queue 2 (1/4), LoadSingle
[02:47] [debug]   PRI 2 elt 0 LoadSingle @ #x141A227CD40
[02:47] [debug]   PRI 3 elt 0 QueueVAG @ #x7FF70B4C2430
[02:47] [debug] [OVERLORD] FS Open FUCV7   STR
[02:47] [debug] [OVERLORD] FS_Close FUCV7   STR @ 136/97
[02:47] [debug]   PRI 3 elt 0 QueueVAG @ #x7FF70B4C2430
[02:47] [info] link finish: fuel-cell-victory-7
****** CALL TO kopen() ******
pc settings file write: "C:\Users\NinjaPC\AppData\Roaming\OpenGOAL\jak1\settings/pc-settings.gc"
****** CALL TO kopen() ******
pc settings file write: "C:\Users\NinjaPC\AppData\Roaming\OpenGOAL\jak1\settings/pc-settings.gc"
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
2
GAMEPLAY: enter training
2
2
2
GAMEPLAY: enter training
2 ```


`(format #t "~D~%" (-> *game-info* money-per-level (level-task-data-index training)))`


```(when (and *target* (= *master-mode* 'game) (not  (zero? (- (current-time) (-> *game-info* blackout-time)))))
   (update-autosplit-info-jak1)
    )``` 
    None of these has seemed to help yet

**Comment by Zedb0T** [comment]
I have this fixed locally will open a PR soon

**Comment by Zedb0T** [comment]
Fixed in #3297 https://github.com/open-goal/jak-project/pull/3297
